File: backend/app/src/main/scala/com/github/ai/leetcodequiz/entity/Questionnaire.scala
package com.github.ai.leetcodequiz.entity

import com.github.ai.leetcodequiz.data.db.model.{QuestionUid, QuestionnaireUid}

case class Questionnaire(
  uid: QuestionnaireUid,
  questions: List[QuestionUid],
  answers: List[Answer],
  isComplete: Boolean
)

File: android/app/src/main/java/com/aivanovski/leetcode/android/entity/Questionnaire.kt
package com.aivanovski.leetcode.android.entity

import com.github.ai.leetcodequiz.api.QuestionnaireStatsDto

data class Questionnaire(
    val id: String,
    val isComplete: Boolean,
    val questions: List<Question>,
    val answers: List<QuestionAnswer>,
    // TODO: create data class
    val stats: QuestionnaireStatsDto
)